shared/src/business/useCases/filePetitionInteractor.js

```diff
@@ -78,11 +78,19 @@ exports.filePetitionInteractor = async ({
     stinFileUpload,
   ]);
 
-  return await applicationContext.getUseCases().createCaseInteractor({
-    applicationContext,
-    ownershipDisclosureFileId: await ownershipDisclosureFileUpload,
-    petitionFileId: await petitionFileUpload,
-    petitionMetadata,
-    stinFileId: await stinFileUpload,
-  });
+  const stinFileId = await stinFileUpload;
+  const caseDetail = await applicationContext
+    .getUseCases()
+    .createCaseInteractor({
+      applicationContext,
+      ownershipDisclosureFileId: await ownershipDisclosureFileUpload,
+      petitionFileId: await petitionFileUpload,
+      petitionMetadata,
+      stinFileId: await stinFileUpload,
+    });
+
+  return {
+    caseDetail,
+    stinFileId,
+  };
```

Comment:
While this functions correctly, there’s some goofy syntax going on with `await`-ing on resolved promises — by the time we pass `await Promise.all(…)` on line 75, we have already resolved all the above promises. Instead, we can capture the values from line 75 and use it below: 

```javascript
const [ ownershipDisclosureFileId, petitionFileId, stinFileId ] = await Promise.all([
  ownershipDisclosureFileUpload,
  petitionFileUpload, 
  stinFileUpload,
]);
```

Reply:
Totally agree. I started to go down this path, but went too far. I've simplified based on your suggestion. Totally comfortable that await `<undefined>` still resolves to `<undefined>`. Much more readable head to toe. 🙏